configs/configupgrade/test-fixtures/valid/indexed-splat/input/indexed-splat.tf

```diff
@@ -0,0 +1,12 @@
+resource "test_instance" "first_many" {
+  count = 2
+}
+
+resource "test_instance" "one" {
+  image = "${test_instance.first_many.*.id[0]}"
+}
+
+resource "test_instance" "second_many" {
+  count = "${length(test_instance.first_many)}"
+  security_groups = "${test_instance.first_many.*.id[count.index]}"
+}
```

Comment:
Could you also include in here something like the following?

```hcl
resource "test_instance" "splat_of_one" {
  image = "${test_instance.one.*.id[0]}"
}
```

For correct results, we'd need to write this one more conservatively:

```hcl
resource "test_instance" "splat_of_one" {
  image = test_instance.one.*.id[0]
}
```

...because applying `.*` when `count` isn't set is permitted in Terraform 0.11 and produces a single-item list, which is preserved in HCL 2, but you can't apply the _index_ operator to a single object, so a full rewrite to the new-style index syntax would break a working config.

I hadn't implemented this rewrite because the current behavior produced a working-but-not-ideal config and that seemed good enough for now. With that said, I _think_ we have enough information available in the analysis object to decide whether the resource has count set, so we could potentially make this work by checking that first.

Reply:
Done.